pkg/osbuild/bootupd_stage.go

```diff
@@ -76,17 +77,56 @@ func NewBootupdStage(opts *BootupdStageOptions, devices map[string]Device, mount
 	}, nil
 }
 
-func GenBootupdDevicesMounts(filename string, pt *disk.PartitionTable) (map[string]Device, []Mount) {
-	_, mounts, devices, err := genMountsDevicesFromPt(filename, pt)
-	if err != nil {
-		panic(err)
+func genMountsForBootupd(source string, pt *disk.PartitionTable) ([]Mount, error) {
+	mounts := make([]Mount, 0, len(pt.Partitions))
+	// note that we are not using pt.forEachMountable() here because we
+	// need to keep track of the partition number (even if it's not
+	// mountable)
+	for idx, part := range pt.Partitions {
+		if part.Payload == nil {
+			continue
+		}
+		// TODO: support things like LVM here via supporting "disk.Container"
```

Comment:
I'm a bit uncertain longer term about having custom code for stuff like that versus using anaconda to initialize LVM.